consensus/istanbul/proxy/proxied_validator_engine.go

```diff
@@ -89,52 +112,104 @@ func NewProxiedValidatorEngine(backend BackendForProxiedValidatorEngine, config
 		config:  config,
 		logger:  log.New(),
 		backend: backend,
-	}
 
-	pv.ph = newProxyHandler(backend, pv)
+		addProxies:      make(chan []*istanbul.ProxyConfig),
+		removeProxies:   make(chan []*enode.Node),
+		addProxyPeer:    make(chan consensus.Peer, 10),
+		removeProxyPeer: make(chan consensus.Peer, 10),
+
+		proxiedValThreadOpCh:     make(chan proxiedValThreadOpFunc),
+		proxiedValThreadOpDoneCh: make(chan struct{}),
+
+		sendValEnodeShareMsgsCh: make(chan struct{}),
+		newBlockchainEpoch:      make(chan struct{}),
+	}
 
 	return pv, nil
 }
 
 func (pv *proxiedValidatorEngine) Start() error {
-	if err := pv.ph.Start(); err != nil {
-		return err
+	pv.isRunningMu.Lock()
+	defer pv.isRunningMu.Unlock()
+
+	if pv.isRunning {
+		return istanbul.ErrStartedProxiedValidatorEngine
 	}
 
+	pv.loopWG.Add(1)
+	pv.quit = make(chan struct{})
+	go pv.threadRun()
+
 	if len(pv.config.ProxyConfigs) > 0 {
-		pv.ph.addProxies <- pv.config.ProxyConfigs
+		pv.addProxies <- pv.config.ProxyConfigs
```

Comment:
This, and all other sends to the member channels of this object, should also `select` on the `pv.quit` channel for extra safety. I saw that you often will check if the engine is stopped at the start of the function. Was that added to handle this risk? If so, I don't think that is a strong enough guarantee that there will not be a race condition, especially because some of the function trigger from the RPC API thread, which may execute at any time.